tests/functional/test_cache.py

```diff
@@ -216,3 +216,15 @@ def test_cache_purge_too_many_args(script, wheel_cache_files):
     # Make sure nothing was deleted.
     for filename in wheel_cache_files:
         assert os.path.exists(filename)
+
+
+@pytest.mark.parametrize("command", ["info", "list", "remove", "purge"])
+def test_cache_abort_when_no_cache_dir(script, command):
+    """Running any pip cache command when cache is disabled should
+    abort and print informative answer"""
```

Comment:
Nit: `abort and log an error` instead of `abort and print informative answer`

Reply:
Take a look now @deveshks

Reply:
Yep, looks good 😊